scrollingelement.js

```diff
@@ -36,12 +36,15 @@ if (!('scrollingElement' in document)) (function() {
 		return null;
 	}
 
+	function isStandardsMode() {
+		return /^CSS1/.test(document.compatMode);
+	}
+
 	// Note: standards mode / quirks mode can be toggled at runtime via
 	// `document.write`.
 	var isCompliantCached;
 	var isCompliant = function() {
-		var isStandardsMode = /^CSS1/.test(document.compatMode);
-		if (!isStandardsMode) {
+		if (!isStandardsMode()) {
```

Comment:
Because of the comment above indicating that the mode can change at runtime, saving into a variable like is done with `isCompliantCached` would only work because `isCompliant()` is always called right before `isStandardsMode()`. This feels fragile.

I can ask the caller to pass in the value of the standards mode check into the `isCompliant()` function, which will let the caller cache this value local to its scope. Thoughts?